extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSources.java

```diff
@@ -0,0 +1,116 @@
+     * and {@link #builder()}
+     */
+    @Nonnull
+    public static BatchSource<String> elastic() {
+        return elastic(ElasticClients::client);
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     * Queries all indexes for all documents.
+     * Uses {@link SearchHit#getSourceAsString()} as mapping function
+     */
+    @Nonnull
+    public static BatchSource<String> elastic(@Nonnull SupplierEx<RestHighLevelClient> clientSupplier) {
+        return elastic(clientSupplier, SearchHit::getSourceAsString);
+    }
+
+    /**
+     * Creates a source which queries local instance of Elasticsearch for all documents
+     * Uses {@link SearchHit#getSourceAsString()} as mapping function
+     */
+    @Nonnull
+    public static <T> BatchSource<T> elastic(@Nonnull FunctionEx<? super SearchHit, T> mapHitFn) {
+        return elastic(ElasticClients::client, mapHitFn);
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     * Uses provided mapHitFn to map results.
+     * Queries all indexes for all documents.
+     *
+     * @param clientSupplier RestHighLevelClient supplier
+     * @param mapHitFn       supplier of a function mapping the result from SearchHit to a result type
+     * @param <T>            result type returned by the map function
+     */
+    @Nonnull
+    public static <T> BatchSource<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
+            @Nonnull FunctionEx<? super SearchHit, T> mapHitFn) {
+        return elastic(clientSupplier, SearchRequest::new, mapHitFn);
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     *
+     * @param clientSupplier        RestHighLevelClient supplier
+     * @param searchRequestSupplier supplier of a SearchRequest used to query for documents
+     * @param mapHitFn              supplier of a function mapping the result from SearchHit to a target type
+     * @param <T>                   result type returned by the map function
+     */
+    @Nonnull
+    public static <T> BatchSource<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
+            @Nonnull SupplierEx<SearchRequest> searchRequestSupplier,
+            @Nonnull FunctionEx<? super SearchHit, T> mapHitFn
+    ) {
+        return ElasticSources.<T>builder()
+                .clientSupplier(clientSupplier)
+                .searchRequestSupplier(searchRequestSupplier)
+                .mapHitFn(mapHitFn)
+                .build();
+    }
+
+    /**
+     * Returns {@link ElasticSourceBuilder}
+     *
+     * @param <T> result type returned by the map function
+     */
+    @Nonnull
+    public static <T> ElasticSourceBuilder<T> builder() {
```

Comment:
this requires you declare `T` upfront. normally we start with `Void` and then map it when map function is defined

Reply:
I tried to mimic what's in SourceBuilder, could you please check this is what you had in mind? (I did the same for ElasticSinkBuilder as well).